tools/psa/__init__.py

```diff
@@ -0,0 +1,17 @@
+from os.path import basename
+
+
+# Find secure image.
+def find_secure_image(toolchain, resources, ns_image_path, configured_s_image_path, image_type):
+    # Locate user-specified image
+    image_files = resources.get_file_paths(image_type)
+    secure_image = next((f for f in image_files if basename(f) == configured_s_image_path), None)
+    secure_image = next((f for f in image_files if basename(f) == basename(ns_image_path)), secure_image)
+
+    if secure_image:
+        toolchain.notify.debug("Secure image file found: %s." % secure_image)
+    else:
+        toolchain.notify.debug("Secure image file %s not found. Aborting." % configured_s_image_path)
```

Comment:
Resources has a notifier too, and if you're only using the notifier from the toolchain, pass in the notifier please.

Reply:
Ok